cpp/lib/src/MediaTypeUtil.cc

```diff
@@ -114,6 +115,24 @@ std::string GetMediaType(const std::string &document, const bool auto_simplify)
 }
 
 
+std::string GetMediaType(const std::string &document, std::string * const subtype, const bool auto_simplify) {
+    std::string media_type = GetMediaType(document, auto_simplify);
+
+    if (media_type == "text/xml") {
+        XMLParser parser(document, XMLParser::XML_STRING);
+
+        std::map<std::string, std::string> attrib_map;
+        XMLParser::XMLPart part;
+        if (parser.skipTo(XMLParser::XMLPart::OPENING_TAG, "", &part)) {
+            if (part.data_ == "TEI")
```

Comment:
This strategy seems a bit risky.  Can't we use a more robust way of determining the subtype?

Reply:
now also checking xmlns